### DISCUSSION TITLE
How could I use whisper to transcribe long video locally on whisper V3?

### DISCUSSION THREAD

**Discussion by jasonsu123** [openai/whisper#1846] [Q&A]
Hello everyone, I currently want to use Whisper for speech synthesis in videos, but I’ve encountered a few issues.

I am a Plus user, and I’ve used the paid API to split a video into one file per minute and then batch process. 

But later I found out that the latest Whisper V3 can only be used locally and cannot be called via API?

I have some knowledge of Python programming. If I want to write a program on my computer, do you have any examples I can refer to?
My speech video is approximately 40 minutes long and over 500MB. I used the Whisper API before, but due to file size limitations, I had to split it for batch processing.

If I use Whisper locally on my computer, will there be no file size limitations?

Also, how can I find the names of different model types to use in the code, such as base, medium, large, and the latest V3?

Thank you.

**Comment by glangford** [comment]
To install `whisper` locally, start with the README document here which contains the information you need and the names of the models.
https://github.com/openai/whisper/blob/main/README.md

For local processing of a 500MB file, there is no need to split it up. Just provide the name of the file to `whisper` in the command line. As long as `ffmpeg` can process the file, whisper should be able to process the audio.

If you have questions, be sure to search the past discussions for any hints and check out `whisper --help`.

**Reply by mjyoussef** [reply]
Just as a follow-up, does model load the entire file into memory before transcribing or does it using streaming?

**Reply by ryanheise** [reply]
It loads the entire file into memory, although there is a PR to load in chunks here: #2256
